src/transport/RendezvousSession.cpp

```diff
@@ -202,6 +202,7 @@ void RendezvousSession::OnRendezvousConnectionClosed()
 
 void RendezvousSession::OnRendezvousError(CHIP_ERROR err)
 {
+    ReleasePairingSessionHandle();
```

Comment:
would it make sense to place this in UpdateState? It seems we want a cleared session whenever we kInit or kSecurePairing.